frontend/src/toolbar/elements/heatmapToolbarMenuLogic.ts

```diff
@@ -475,10 +503,28 @@ export const heatmapToolbarMenuLogic = kea<heatmapToolbarMenuLogicType>([
 
         // Store for cleanup
         cache.intersectionObserver = intersectionObserver
+
+        cache.mutationObserver = new MutationObserver(() => {
+            const scrollY = values.posthog?.scrollManager?.scrollY() ?? 0
+            if (values.heatmapScrollY !== scrollY) {
+                actions.setHeatmapScrollY(scrollY)
+            }
+        })
```

Comment:
this makes me wonder... does the mutation observer fire for all scrolls?

my first thought is always "can we avoid adding _this_"
every line we add is one that needs maintaining

do we need a scroll listener and a mutation observer?